pkg/controllers/monitoring/monitoring-stack/conditions.go

```diff
@@ -0,0 +1,125 @@
+package monitoringstack
+
+import (
+	"fmt"
+
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/rhobs/observability-operator/pkg/apis/monitoring/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	AvailableReason                = "MonitoringStackAvailable"
+	ReconciledReason               = "MonitoringStackReconciled"
+	PrometheusNotAvailable         = "PrometheusNotAvailable"
+	PrometheusNotReconciled        = "PrometheusNotReconciled"
+	CannotReadPrometheusConditions = "Cannot read Prometheus status conditions"
+	AvailableMessage               = "Monitoring Stack is available"
+	ReconciledMessage              = "Monitoring Stack successfully reconciled"
+)
+
+func updateConditions(msConditions []v1alpha1.Condition, promConditions []monv1.PrometheusCondition, generation int64, recError error) []v1alpha1.Condition {
+	if len(msConditions) == 0 {
```

Comment:
when would we enter this branch?

Reply:
I think in the very first reconciliation when there's already `MonitoringStack` available, but with no conditions.